cmd/snap-bootstrap/cmd_initramfs_mounts_test.go

```diff
@@ -813,3 +817,54 @@ func (s *initramfsMountsSuite) TestInitramfsMountsRunModeKernelStatusTryingNoTry
 	c.Check(s.Stdout.String(), Equals, fmt.Sprintf(`%[1]s/ubuntu-data/system-data/var/lib/snapd/snaps/pc-kernel_1.snap %[1]s/kernel
 `, boot.InitramfsRunMntDir))
 }
+
+func (s *initramfsMountsSuite) TestUnlockEncryptedPartition(c *C) {
+	tcti, err := os.Open("/dev/null")
+	c.Assert(err, IsNil)
+	tpm, err := tpm2.NewTPMContext(tcti)
+	c.Assert(err, IsNil)
+	mockTPM := &secboot.TPMConnection{TPMContext: tpm}
+	restoreConnect := main.MockSecbootConnectToDefaultTPM(func() (*secboot.TPMConnection, error) {
+		return mockTPM, nil
+	})
+	defer restoreConnect()
+
+	tc := func(activationSuccessful bool, activationError error) error {
```

Comment:
I implemented it as a function instead of a iteractive table to avoid nesting deferred mock restores in a loop, but I agree that the table for is the standard one and will refactor as such.

Reply:
Will also refactor a similar construct in #8463